apps/constellation/key_generator.cpp

```diff
@@ -0,0 +1,108 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "crypto/ecdsa.hpp"
+
+#include <cstdlib>
+#include <fstream>
+
+namespace fetch {
+
+using fetch::crypto::Prover;
+using fetch::crypto::ECDSASigner;
+
+using SignerPtr = std::shared_ptr<ECDSASigner>;
+using ProverPtr = std::shared_ptr<Prover>;
+
+char const *LOGGING_NAME         = "KeyGenerator";
+char const *DEFAULT_KEY_FILENAME = "p2p.key";
```

Comment:
Probably meaningless, but these may be constexpr

Reply:
`DEFAULT_KEY_FILENAME` certainly can't because we need to be able to make a pointer to it! 😁